packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/workers/downloadTrackWorker.ts

```diff
@@ -157,13 +160,36 @@ function* downloadTrackAudio(track: UserTrackMetadata) {
   const trackFilePath = getLocalAudioPath(track_id)
   const encodedTrackId = encodeHashId(track_id)
 
-  for (const creatorNodeEndpoint of creatorNodeEndpoints) {
-    const trackAudioUri = `${creatorNodeEndpoint}/tracks/stream/${encodedTrackId}`
-    const response = yield* call(downloadFile, trackAudioUri, trackFilePath)
-    const { status } = response.info()
-    if (status === 200) return
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  const apiClient = yield* getContext('apiClient')
+  const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+  const isGatedContentEnabled = yield* call(() => {
+    return getFeatureEnabled(FeatureFlags.GATED_CONTENT_ENABLED)
+  })
+  const queryParams: QueryParams = { filename: `${track_id}.mp3` }
+  if (isGatedContentEnabled) {
+    const data = `Premium content user signature at ${Date.now()}`
+    const signature = yield* call(audiusBackendInstance.getSignature, data)
+    const premiumTrackSignatureMap = yield* select(getPremiumTrackSignatureMap)
+    const premiumContentSignature =
+      premium_content_signature || premiumTrackSignatureMap[track_id]
+    queryParams.user_data = data
+    queryParams.user_signature = signature
+    if (premiumContentSignature) {
+      queryParams.premium_content_signature = JSON.stringify(
+        premiumContentSignature
+      )
+    }
   }
 
+  const trackAudioUri = apiClient.makeUrl(
+    `/tracks/${encodedTrackId}/stream`,
+    queryParams
+  )
+  const response = yield* call(downloadFile, trackAudioUri, trackFilePath)
+  const { status } = response.info()
+  if (status === 200) return
```

Comment:
super nit but "don't special-case the happy path" is something I've tried to adhere to. In this case maybe `if (status !== 200) throw error else return`, but up to you.

Reply:
fair. i did some copy/pasta :P